javascript/javascript-import-and-require.md

```diff
@@ -0,0 +1,149 @@
+# JavaScript - Require & Import
+
+## Project Time
+
+About 40 Minutes
+
+- video and reference material: 20 minutes
+- Understanding topics: 10 minutes 
+- Pratice challenge: 10 minutes
+
+## Prerequisites
+
+- JavaScript 7
+
+## Motivation
+
+Javascript is one of the most used language. It is lightweight, Interpreted and Just-in-time compiled programming language with the first-class function. While it is most well-known as the scripting language for a webpage, Many no-browser environments also use it. JavaScript is a prototype-based, multi-paradigm, single-threaded, dynamic language, supporting object-oriented, imperative and declarative style.
+
+Here in Javascript, Import and Require is one of the important part and  also crucial part of it. It major useful function, It help us to divide the files into different modules and divide our code modules in systematic ways.
+
+```
+Porject:
+        assert
+          |
+          --> index.js
+        img
+          |
+          --> file.png
+        css
+          |
+          --> index.css
+          
+        index.html
+```
+Now you will get one question in your mind that How we use index.js in index.html, Now Import and Require will come is use here and using these function, you can organize your work in systematic way.
+
+## Objectives
+
+- Create understanding of import and require in user
+
+## Supplemental Materials
+
+- [JavaScript reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference)
+- [JavaScript Modules: ES6 Import and Export](https://www.youtube.com/watch?v=_3oSWwapPKQ)
+- [JavaScript Import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import)
+- [JavaScript Require](https://nodejs.org/en/knowledge/getting-started/what-is-require/)
+- [JavaScript webpack resolver](https://webpack.js.org/concepts/module-resolution/)
+
+## Lesson
+
+**Why REQUIRE and IMPORT are important in Javascript?**
+
+Code is easier to manage when it is in small, bite-size chunks. This is the thinking behind keeping functions to only one task or having files contain only a few, or one, component at a time. Here both import and require play major role in minimize and allow us to handle large number of module effectively and efficently. It is mainly used to get modules from different dictionary and use them. If you have a complex app and have to scroll through hundreds or thousands of lines of code, then the job of debugging or just understanding the app becomes that much harder.

```

Comment:
Did you take this text from here?: https://medium.com/@thejasonfile/a-simple-intro-to-javascript-imports-and-exports-389dd53c3fac